ethcore/src/client/client.rs

```diff
@@ -660,6 +672,51 @@ impl BlockChainClient for Client {
 		ret
 	}
 
+	fn replay(&self, id: TransactionID, analytics: CallAnalytics) -> Result<Executed, ReplayError> {
+		if let Some(address) = self.transaction_address(id) {
```

Comment:
How about this:
```rust
let address = try!(self.transaction_address(id).ok_or(ReplayError::TransactionNotFound));
let block_data = try!(self.block(BlockID::Hash(address.block_hash)).ok_or(ReplayError::StatePruned));
let mut state = try!(self.state_at_beginning(BlockID::Hash(address.block_hash).ok_or(ReplayError::StatePruned));

let block = ...

if address.index >= txs.len() {
  return Err(ReplayError::TransactionNotFound);
}
...
```

to avoid nesting?